pkg/executor/aggfuncs/spill_deserialize_helper.go

```diff
@@ -0,0 +1,436 @@
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeTypesDuration(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinString(dst *partialResult4MaxMinString) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeString(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinJSON(dst *partialResult4MaxMinJSON) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeBinaryJSON(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinEnum(dst *partialResult4MaxMinEnum) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeEnum(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinSet(dst *partialResult4MaxMinSet) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeSet(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4AvgDecimal(dst *partialResult4AvgDecimal) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.sum = spill.DeserializeMyDecimal(bytes, &pos)
+		dst.count = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4AvgFloat64(dst *partialResult4AvgFloat64) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.sum = spill.DeserializeFloat64(bytes, &pos)
+		dst.count = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4SumDecimal(dst *partialResult4SumDecimal) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.val = spill.DeserializeMyDecimal(bytes, &pos)
+		dst.notNullRowCount = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4SumFloat64(dst *partialResult4SumFloat64) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
```

Comment:
It looks to me that every `deserializePartialXXX` will have the same code like 
```
pos := int64(0)
bytes := s.column.GetBytes(s.readRowIndex)
```
How about wrap a new struct that contains `pos` and `bytes`, add add a new function in `spillDeserializeHelper` to create such a struct, and in this case, all the `spill.DeserializeXXX(bytes, &pos)` can also be simplified to `spill.DeserializeXXX(new_struct)`, which mean you don't need to maintain `pos` explicitly.

Reply:
> It looks to me that every `deserializePartialXXX` will have the same code like
> 
> ```
> pos := int64(0)
> bytes := s.column.GetBytes(s.readRowIndex)
> ```
> 
> How about wrap a new struct that contains `pos` and `bytes`, add add a new function in `spillDeserializeHelper` to create such a struct, and in this case, all the `spill.DeserializeXXX(bytes, &pos)` can also be simplified to `spill.DeserializeXXX(new_struct)`, which mean you don't need to maintain `pos` explicitly.

I will wrap these two fields into a new struct. However, I think we should not change the function signature of `spill.DeserializeXXX` as other package may use it.

Reply:
I don't get your point, in my understanding, the `spill.DeserializeXXX` is actually newly added, why there will be other package that use it?

Reply:
> I don't get your point, in my understanding, the `spill.DeserializeXXX` is actually newly added, why there will be other package that use it?

done